tests/cache/tests.py

```diff
@@ -1188,25 +1188,24 @@ def test_memcached_uses_highest_pickle_version(self):
             self.assertEqual(caches[cache_key]._cache.pickleProtocol,
                              pickle.HIGHEST_PROTOCOL)
 
-    @override_settings(CACHES=caches_setting_for_tests(
-        base=memcached_params,
-        exclude=memcached_excluded_caches,
-        TIMEOUT=None,
-    ))
     def test_default_never_expiring_timeout(self):
         # Regression test for #22845
-        cache.set('infinite_foo', 'bar')
-        self.assertEqual(cache.get('infinite_foo'), 'bar')
+        with override_settings(CACHES=caches_setting_for_tests(
```

Comment:
Combining this commit with the next one seems fine. It works the same as override_settings, but you can use `with self.settings`.